roles/wordpress-setup/tasks/self-signed-certificate.yml

```diff
@@ -1,10 +1,19 @@
 ---
 - name: Generate self-signed certificates
-  shell: >
-    openssl req -subj "/CN={{ item.value.site_hosts[0].canonical }}" -new
-    -newkey rsa:2048 -days 3650 -nodes -x509 -sha256
-    -keyout {{ item.key }}.key -out {{ item.key }}.cert
+  shell: "openssl req -subj \"/CN={{ item.value.site_hosts[0].canonical }}\" -new \
+    -newkey rsa:2048 -days 3650 -nodes -x509 -sha256 \
+    -extensions req_ext -config <( \
+cat <<\" EOF\"\n
+[req]\n
+distinguished_name = dn\n
+[ dn ]\n
+[ req_ext ]\n
+subjectAltName = {{ site_hosts | union(multisite_subdomains_wildcards) | map('regex_replace', '(.*)', 'DNS:\\1') | join(',') }}\n
+EOF\n
+    ) \
+    -keyout {{ item.key | quote }}.key -out {{ item.key | quote }}.cert"
   args:
+    executable: "/bin/bash"
```

Comment:
The bash shell has to be used, since virtual files use the subshell operator `()` (l.5), which produces a syntax error using the default `sh` shell.
Perhaps there is a way to make this work with sh, but i didn't find it.